lib/async/redis/sentinels.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+module Async
+	module Redis
+		class SentinelsClient < Client
+			def initialize(master_name, sentinels, role = :master, protocol = Protocol::RESP2, **options)
+				@master_name = master_name
+				@sentinel_endpoints = sentinels.map do |sentinel|
+					Async::IO::Endpoint.tcp(sentinel[:host], sentinel[:port])
+				end
+				@role = role
+
+				@protocol = protocol
+				@pool = connect(**options)
+			end
+
+			private
+
+			# Override the parent method. The only difference is that this one needs
+			# to resolve the master/slave address.
+			def connect(**options)
+				Async::Pool::Controller.wrap(**options) do
+					addr = resolve_addr
+					endpoint = Async::IO::Endpoint.tcp(addr[:ip], addr[:port])
+
+					peer = endpoint.connect
+
+					peer.setsockopt(Socket::IPPROTO_TCP, Socket::TCP_NODELAY, 1)
+
+					stream = IO::Stream.new(peer)
+
+					@protocol.client(stream)
+				end
+			end
+
+			def resolve_addr
+				addr = case @role
+				when :master
+					resolve_master
+				when :slave
+					resolve_slave
+				else
+					raise ArgumentError, "Unknown instance role #{@role}"
+				end
+
+				addr || (raise RuntimeError, "Unable to fetch #{@role} via Sentinel.")
```

Comment:
I think if you use `address or raise ...` it doesn't need brackets and is slightly more canonical.

Reply:
Done :+1: